### PR TITLE
Validate Security.* endpoints

### PR DESCRIPTION
**Pull request by philkra**
should provide full coverage to security.

* `security.put_user` request
* `security.put_role` request
* `security.get_role` response
* `security.get_privileges` request
* `security.create_api_key` request
* `security.authenticate` response

### CODE REVIEW

**Review by Mpdreamz — DISMISSED** [review]
LGTM

File: specification/specs/x_pack/security/privileges/get_privileges/GetPrivilegesRequest.ts [hunk 1]
```diff
@@ -24,8 +24,8 @@
  */
 interface GetPrivilegesRequest extends RequestBase {
   path_parts?: {
-    application?: Name
-    name?: Name
+    application?: Name | boolean
+    name?: Name | boolean
```
Comment (delvedor) [inline specification/specs/x_pack/security/privileges/get_privileges/GetPrivilegesRequest.ts:28]: This looks very weird, the [resta-api-spec](https://github.com/elastic/elasticsearch/blob/7.x/x-pack/plugin/src/test/resources/rest-api-spec/api/security.get_privileges.json) never mentions the boolean type, nor the [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-get-privileges.html).

I tried to run this test on my machine, but it passed. Can you paste here the filename of the failing test?
I guess that the yaml tests are doing something weird.

**Comment by philkra** [comment]
@delvedor , I changed the specification/specs/x_pack/security/privileges/get_privileges/GetPrivilegesRequest.ts back to
```
path_parts?: {
    application?: Name
    name?: Name
  }
```
removing `boolean` union with `Name`. The following tests fail:
```
workbench/xpack-Test_put_and_delete_privileges_5_request.json.test-d.ts:4:2
  ✖  4:2  Type true is not assignable to type string | undefined.  
  ✖  5:2  Type true is not assignable to type string | undefined.  

  workbench/xpack-Test_put_and_get_privileges_7_request.json.test-d.ts:4:2
  ✖  4:2  Type true is not assignable to type string | undefined.  
  ✖  5:2  Type true is not assignable to type string | undefined.  

  workbench/xpack-Test_put_and_get_privileges_6_request.json.test-d.ts:5:2
  ✖  5:2  Type true is not assignable to type string | undefined. 
```

content of `xpack-Test_put_and_delete_privileges_5_request.json.test-d.ts`:
```
import { expectAssignable } from 'tsd'
import * as T from '../types'
expectAssignable<T.GetPrivilegesRequest>({
  "application": true,
  "name": true
})
```

**Comment by philkra** [comment]
@delvedor pushed https://github.com/elastic/elastic-client-generator/pull/206/commits/1b5d053634dfd375cacb82f8f52c92bd4cd896ce to remove the union as it will be addressed with https://github.com/elastic/clients-flight-recorder/pull/669

**Review by delvedor — APPROVED** [review]

**Pull request merged**